File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area = 0
        left, right = 0, len(height) - 1
        while (width := right - left) > 0:
            if height[left] < height[right]:
                area = max(area, width * height[left])
                left += 1
            else:
                area = max(area, width * height[right])
                right -= 1
        return area

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int result = 0;
        int area;

        while (left < right) {
            area = (right - left) * min(height[left], height[right]);
            result = max(result, area);

            if (height[left] < height[right]) {
                left += 1;
            } else {
                right -= 1;
            }
        }
        return result;
    }
};
